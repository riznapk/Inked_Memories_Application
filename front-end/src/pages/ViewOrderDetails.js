import React from "react";

function ViewOrderDetails({ data }) {
  return (
    <div>
      <h2>Order Details</h2>
      <p>Order ID: {data.orderID}</p>
      <p>Order Placed Date: {data.orderPlacedDate}</p>
      <p>Status: {data.status}</p>
      <p>Payment: {data.payment}</p>
      <p>Amount: £{data.amount.toFixed(2)}</p>

      <h3>Cart Items: </h3>
      {data.cartItems.map((item, index) => (
        <div key={index} style={{ marginTop: "30px", marginBottom: "30px" }}>
          <img
            src={item.designImageURI}
            style={{ objectFit: "contain", height: "180px", width: "180px" }}
          />
          <p>Cart ID: {item.cartID}</p>
          <p>Design Font: {item.designFont}</p>
          <p>Design Price: £{item.designPrice}</p>
          <p>Message: {item.message.split(" ").slice(0, 20).join(" ")}</p>
          <p>Scheduled Date: {item.scheduledDate}</p>
          {item?.isLetterIncluded ? <p>Letter: Included with the card</p> : ""}

          <h4>Address: </h4>
          <p>Street Address: {item.address.streetAddress}</p>
          <p>City: {item.address.city}</p>
          <p>County: {item.address.county}</p>
          <p>Postal Code: {item.address.postalCode}</p>
        </div>
      ))}
    </div>
  );
}

export default ViewOrderDetails;
