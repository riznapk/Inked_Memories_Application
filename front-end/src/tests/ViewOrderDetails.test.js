import React from "react";
import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import renderer from "react-test-renderer";
import ViewOrderDetails from "../pages/ViewOrderDetails";
import { MemoryRouter } from "react-router-dom";
test("View order details", () => {
  const data = {
    orderID: "123456",
    orderPlacedDate: "2023-08-06",
    status: "Processing",
    payment: "Credit Card",
    amount: 99.99,
    cartItems: [
      {
        designImageURI: "design-image.jpg",
        cartID: "cart123",
        designFont: "Arial",
        designPrice: 49.99,
        message: "Test message for snapshot testing",
        scheduledDate: "2023-08-10",
        isLetterIncluded: true,
        address: {
          streetAddress: "123 Main St",
          city: "Cityville",
          county: "Countyshire",
          postalCode: "12345",
        },
      },
    ],
  };
  const component = renderer.create(
    <MemoryRouter>
      <ViewOrderDetails data={data} />
    </MemoryRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
