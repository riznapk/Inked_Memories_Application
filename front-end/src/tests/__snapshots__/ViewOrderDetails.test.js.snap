// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View order details 1`] = `
<div>
  <h2>
    Order Details
  </h2>
  <p>
    Order ID: 
    123456
  </p>
  <p>
    Order Placed Date: 
    2023-08-06
  </p>
  <p>
    Status: 
    Processing
  </p>
  <p>
    Payment: 
    Credit Card
  </p>
  <p>
    Amount: £
    99.99
  </p>
  <h3>
    Cart Items: 
  </h3>
  <div
    style={
      {
        "marginBottom": "30px",
        "marginTop": "30px",
      }
    }
  >
    <img
      src="design-image.jpg"
      style={
        {
          "height": "180px",
          "objectFit": "contain",
          "width": "180px",
        }
      }
    />
    <p>
      Cart ID: 
      cart123
    </p>
    <p>
      Design Font: 
      Arial
    </p>
    <p>
      Design Price: £
      49.99
    </p>
    <p>
      Message: 
      Test message for snapshot testing
    </p>
    <p>
      Scheduled Date: 
      2023-08-10
    </p>
    <p>
      Letter: Included with the card
    </p>
    <h4>
      Address: 
    </h4>
    <p>
      Street Address: 
      123 Main St
    </p>
    <p>
      City: 
      Cityville
    </p>
    <p>
      County: 
      Countyshire
    </p>
    <p>
      Postal Code: 
      12345
    </p>
  </div>
</div>
`;
